src/System.Device.Gpio.Tests/GpioControllerTestBase.cs

```diff
@@ -130,30 +131,203 @@ public void OpenPinDefaultsModeToInput()
                 controller.OpenPin(OutputPin, PinMode.Output);
                 controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Rising, callback);
                 controller.Write(OutputPin, PinValue.High);
-                mre.WaitOne(TimeSpan.FromSeconds(5));
+                Thread.Sleep(10);
                 Assert.True(wasCalled);
             }
 
             void callback(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
             {
                 wasCalled = true;
-                mre.Set();
+            }
+        }
+
+        [Fact]
+        public void AddCallbackFallingEdgeNotDetectedTest()
+        {
+            bool wasCalled = false;
+            using (GpioController controller = new GpioController(GetTestNumberingScheme(), GetTestDriver()))
+            {
+                controller.OpenPin(InputPin, PinMode.Input);
+                controller.OpenPin(OutputPin, PinMode.Output);
+                controller.Write(OutputPin, PinValue.Low);
+                controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Falling, callback);
+                controller.Write(OutputPin, PinValue.High);
+                Thread.Sleep(10);
+                Assert.False(wasCalled);
+            }
+
+            void callback(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
+            {
+                wasCalled = true;
+            }
+        }
+
+        [Fact]
+        public void AddCallbackRemoveCallbackTest()
+        {
+            int risingEventOccuredCount = 0, fallingEventOccuredCount = 0;
+            using (GpioController controller = new GpioController(GetTestNumberingScheme(), GetTestDriver()))
+            {
+                controller.OpenPin(InputPin, PinMode.Input);
+                controller.OpenPin(OutputPin, PinMode.Output);
+                controller.Write(OutputPin, PinValue.Low);
+
+                controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Rising, (o, e) => {
+                    risingEventOccuredCount++;
+                });                        
+                controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Rising, callback);
+                controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Rising, (o, e) => {
+                    risingEventOccuredCount++;
+                    if (fallingEventOccuredCount == 4)
+                    {
+                        controller.UnregisterCallbackForPinValueChangedEvent(InputPin, callback);
+                    }
+                });
+                controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Falling, (o, e) => {
+                    fallingEventOccuredCount++;
+                });
+
+                for (int i = 0; i < 10; i++)
+                {
+                    controller.Write(OutputPin, PinValue.High);
+                    Thread.Sleep(10);
```

Comment:
same here with these sleep values.

Reply:
Will use same statice variable, suggestion for appropriate value is appreciated